File: app/src/main/java/ch/zuehlke/sbb/reddit/RedditApp.java
package ch.zuehlke.sbb.reddit;

import android.app.Application;

/**
 * Created by chsc on 12.11.17.
 */

public class RedditApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Injection.init();
    }
}

File: app/src/main/java/ch/zuehlke/reddit/RedditApp.kt
package ch.zuehlke.reddit

import android.app.Activity
import android.app.Application
import ch.zuehlke.reddit.di.AppInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import javax.inject.Inject

/**
 * Created by chsc on 12.11.17.
 */

class RedditApp : Application(), HasActivityInjector {

    @Inject
    lateinit var activityInjector: DispatchingAndroidInjector<Activity>

    override fun activityInjector() = activityInjector

    override fun onCreate() {
        super.onCreate()
        AppInjector.init(this)
    }
}
